cloudinit/analyze/show.py

```diff
@@ -360,12 +384,15 @@ def generate_records(
         else:
             prev_evt = unprocessed.pop()
             if event_name(event) == event_name(prev_evt):
-                record = event_record(start_time, prev_evt, event)
-                records.append(
-                    format_record("Finished stage: (%n) %d seconds", record)
-                    + "\n"
-                )
-                total_time += record.get("delta")
+                if start_time:
```

Comment:
Nit: We could do the following and revert the rest of the block.

```suggestion
                if start_time:
                    continue
```